### PR TITLE
fix(blocks): in title line, press key arrowdown will create a new line

### LINKED ISSUE #2807: In title line, press key DownArrow will create a new line
https://github.com/toeverything/blocksuite/assets/41265413/a7da4fe3-c4db-44b8-937a-d0849c97b0d5

### PR DESCRIPTION
**Pull request by donteatfriedrice**
To close #2807 

https://github.com/toeverything/blocksuite/assets/99816898/f5fc6a92-7961-4848-8876-89e45e5757ca

It seems that the first paragraph block is the second child of the page block. So I check the second element of the model's children.

### CODE REVIEW

**Review by Flrande — APPROVED** [review]

**Comment by Flrande** [comment]
Good catch!

**Comment by Flrande** [comment]
It seems that we need to more accurately identify the frame block. Here are some reference codes.

```ts
const firstFrame = model.children.find((block) => block.flavour === "affine:frame")
if (!firstFrame) {
	const newFirstFrameId = page.addBlock("affine:frame", {}, null, 0)
	const newFirstParagraphId = page.addBlock("affine:paragraph", {}, newFirstFrameId, 0)
	asyncFocusRichText(page, newFirstParagraphId)
	return
}
const firstParagraph = firstFrame.children.find((block) => block.flavour === "affine:paragraph")
```

File: packages/blocks/src/page-block/default/default-page-block.ts [hunk 1]
```diff
@@ -248,7 +248,7 @@ export class DefaultPageBlockComponent
       return;
     } else if (e.key === 'ArrowDown' && hasContent) {
       e.preventDefault();
-      const firstParagraph = model.children[0].children[0];
+      const firstParagraph = model.children[1].children[0];
```
Comment (doodlewind) [inline packages/blocks/src/page-block/default/default-page-block.ts:251]: /cc @Saul-Mirone do you have a recommended way to query the first paragraph here?

**Review by doodlewind — APPROVED** [review]
I think this is a good catch that fixes the regression in #2554

File: packages/blocks/src/page-block/default/default-page-block.ts (hunk 1 above)
Comment (Saul-Mirone) [reply packages/blocks/src/page-block/default/default-page-block.ts:251]: Please check the flavour of block.

**Comment by doodlewind** [comment]
@donteatfriedrice I think it's good to merge as long as we get CI passed. Could you take a look a the test cases that failed and possibly do some tweak, so as to land this PR? Thanks!

**Comment by donteatfriedrice** [comment]
@doodlewind Sorry, I have a question about the ArrowDown feature, based on the code logic currently, when press down ArrowDown, the cursor will move from title to the first paragraph, it will ignore the other block types like list. Is this how the feature designed? I attach a video below. Thanks for your reply.

https://github.com/toeverything/blocksuite/assets/99816898/e3feaa82-ebae-4a08-9e1f-8e11da946d30

**Comment by doodlewind** [comment]
@donteatfriedrice good catch! I've added a quick update, feel free to modify if necessary!

**Comment by donteatfriedrice** [comment]
> @donteatfriedrice good catch! I've added a quick update, feel free to modify if necessary!

Thanks! I think maybe it can be merged now..

**Comment by donteatfriedrice** [comment]
@doodlewind I am wondering if it should also support code block?

**Comment by doodlewind** [comment]
> @doodlewind I am wondering if it should also support code block?

Surely nice to have!

**Comment by donteatfriedrice** [comment]
@doodlewind Now it supports ArrowDown from title to paragraph, list, code block. But it seems that there are some problems with the key down events of code block itselef. 
1. When the cursor is in the last code line, user press arrowdown, the cursor would not move to next block.
2. When cursor is at the block before code block, then press arrowdown, it will focus on the whole code block, not the first code line of this code block.
3.  When cursor is at the block after code block, then press arrowup, it will focus on the whole code block, not the last code line of this code block. 

https://github.com/toeverything/blocksuite/assets/99816898/a98f2472-c15c-4e1d-8250-cb9295565bf8

Should these bugs fixed by this issue (#2807), or by issue #2805 ?

**Comment by doodlewind** [comment]
@donteatfriedrice great catch, having more standalone issues and PRs following our discussion will be really appreciated!

**Comment by doodlewind** [comment]
This one is good to merge. I hope we can provide test cases regarding the cursor behavior in the upcoming improvements, thanks!

**Comment by donteatfriedrice** [comment]
> This one is good to merge. I hope we can provide test cases regarding the cursor behavior in the upcoming improvements, thanks!

OK, thanks.

**Pull request merged**